packages/next-swc/crates/core/src/relay.rs

```diff
@@ -0,0 +1,157 @@
+use std::path::PathBuf;
+
+use pathdiff::diff_paths;
+use serde::Deserialize;
+use swc_atoms::{js_word, JsWord};
+use swc_common::FileName::Real;
+use swc_common::{FileName, Span};
+use swc_ecmascript::ast::*;
+use swc_ecmascript::visit::{Fold, FoldWith};
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum RelayLanguageConfig {
+    Typescript,
+    Flow,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Config {
+    pub artifact_directory: Option<PathBuf>,
+    pub language: RelayLanguageConfig,
+}
+
+impl Config {
+    fn file_extension(&mut self) -> &'static str {
+        match self.language {
+            RelayLanguageConfig::Typescript => "ts",
+            RelayLanguageConfig::Flow => "js",
+        }
+    }
+}
+
+struct Relay {
+    config: Config,
+    file_name: FileName,
+}
+
+fn pull_first_operation_name_from_tpl(tpl: TaggedTpl) -> Option<String> {
```

Comment:
Given a graphql template, find the first operation. could be a query, subscription, or mutation